plugins/PrivacyManager/Model/DataSubjects.php

```diff
@@ -114,6 +119,51 @@ public function deleteDataSubjects($visits)
         return $results;
     }
 
+    private function invalidateArchives($visits)
+    {
+        $datesToInvalidateByIdSite = $this->getDatesToInvalidate($visits);
+
+        $invalidator = StaticContainer::get('Piwik\Archive\ArchiveInvalidator');
+
+        foreach ($datesToInvalidateByIdSite as $idSite => $visitDates) {
+            foreach ($visitDates as $dateStr) {
+                $visitDate = Date::factory($dateStr);
+                $invalidator->rememberToInvalidateArchivedReportsLater($idSite, $visitDate);
+            }
+        }
+    }
+
+    private function getDatesToInvalidate($visits)
+    {
+        $idVisitsByIdSites = array();
+        foreach ($visits as $visit) {
+            $idSite = (int)$visit['idsite'];
+            if (!isset($idVisitsByIdSites[$idSite])) {
+                $idVisitsByIdSites[$idSite] = array();
+            }
+            $idVisitsByIdSites[$idSite][] = (int)$visit['idvisit'];
+        }
+
+        $datesToInvalidate = array();
+        foreach ($idVisitsByIdSites as $idSite => $idVisits) {
+            $website = Cache::getCacheWebsiteAttributes($idSite);
```

Comment:
FYI here you can use `\Piwik\Site::getTimezoneFor($idSite)`.  We use `Cache::getCacheWebsiteAttributes` only in tracker mode (it's tracker cache).

Reply:
Thanks, changed.